static/app/views/performance/mobile/components/spanSamplesPanel.tsx

```diff
@@ -82,6 +86,26 @@ export function SpanSamplesPanel({
     })}`
   );
 
+  const handlePrimarySearch = (newSpanSearchQuery: string) => {
+    router.replace({
+      pathname: location.pathname,
+      query: {
+        ...query,
+        primarySpanSearchQuery: newSpanSearchQuery,
+      },
+    });
+  };
+
+  const handleSecondarySearch = (newSpanSearchQuery: string) => {
+    router.replace({
+      pathname: location.pathname,
+      query: {
+        ...query,
+        secondarySpanSearchQuery: newSpanSearchQuery,
+      },
+    });
+  };
```

Comment:
Can we define this as a generic method in the samples container and pass down the key to set in the URL?